src/main/java/uk/gov/hmcts/cmc/claimstore/rules/ClaimantRepaymentPlanRule.java

```diff
@@ -0,0 +1,94 @@
+package uk.gov.hmcts.cmc.claimstore.rules;
+
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.cmc.claimstore.exceptions.ClaimantInvalidRepaymentPlanException;
+import uk.gov.hmcts.cmc.claimstore.utils.CalculateMonthIncrement;
+import uk.gov.hmcts.cmc.domain.models.Claim;
+import uk.gov.hmcts.cmc.domain.models.PaymentOption;
+import uk.gov.hmcts.cmc.domain.models.RepaymentPlan;
+import uk.gov.hmcts.cmc.domain.models.response.FullAdmissionResponse;
+import uk.gov.hmcts.cmc.domain.models.response.PartAdmissionResponse;
+import uk.gov.hmcts.cmc.domain.models.response.Response;
+
+import java.time.LocalDate;
+
+@Service
+public class ClaimantRepaymentPlanRule {
+
+    public void assertClaimantRepaymentPlanIsValid(Claim claim, RepaymentPlan proposedPlan) {
+        Response response = claim.getResponse().orElseThrow(() -> new ClaimantInvalidRepaymentPlanException(
+            ClaimantInvalidRepaymentPlanException.EXPECTED_DEFENDANT_RESPONSE));
+
+        switch (getPaymentOptionForResponse(response)) {
+            case INSTALMENTS:
+                assertInstallmentsIsValidAgainstInstallments(proposedPlan, getRepaymentPlanForResponse(response));
+                break;
+            case BY_SPECIFIED_DATE:
+                assertInstallmentsIsValidAgainstSetDate(proposedPlan);
+                break;
+            default:
+                throw new ClaimantInvalidRepaymentPlanException(
+                    ClaimantInvalidRepaymentPlanException.INVALID_DEFENDANT_REPAYMENT_TYPE);
+        }
+    }
+
+
+    private void assertInstallmentsIsValidAgainstInstallments(RepaymentPlan claimantPlan, RepaymentPlan defendantPlan) {
```

Comment:
rename the method pls. InstallmentsAgainstInstallments for someone who reads this code first time, wont understand what its intent is.